geode-core/src/main/java/org/apache/geode/internal/cache/configuration/CacheConfig.java

```diff
@@ -0,0 +1,3570 @@
+ *         &lt;element name="disk-store" type="{http://geode.apache.org/schema/cache}disk-store-type" maxOccurs="unbounded" minOccurs="0"/>
+ *         &lt;element name="pdx" type="{http://geode.apache.org/schema/cache}pdx-type" minOccurs="0"/>
+ *         &lt;element name="region-attributes" type="{http://geode.apache.org/schema/cache}region-attributes-type" maxOccurs="unbounded" minOccurs="0"/>
+ *         &lt;choice maxOccurs="unbounded" minOccurs="0">
+ *           &lt;element name="jndi-bindings" type="{http://geode.apache.org/schema/cache}jndi-bindings-type"/>
+ *           &lt;element name="region" type="{http://geode.apache.org/schema/cache}region-type"/>
+ *           &lt;element name="vm-root-region" type="{http://geode.apache.org/schema/cache}region-type"/>
+ *         &lt;/choice>
+ *         &lt;element name="function-service" type="{http://geode.apache.org/schema/cache}function-service-type" minOccurs="0"/>
+ *         &lt;element name="resource-manager" type="{http://geode.apache.org/schema/cache}resource-manager-type" minOccurs="0"/>
+ *         &lt;element name="serialization-registration" type="{http://geode.apache.org/schema/cache}serialization-registration-type" minOccurs="0"/>
+ *         &lt;element name="backup" type="{http://www.w3.org/2001/XMLSchema}string" maxOccurs="unbounded" minOccurs="0"/>
+ *         &lt;element name="initializer" type="{http://geode.apache.org/schema/cache}initializer-type" minOccurs="0"/>
+ *         &lt;any processContents='lax' namespace='##other' maxOccurs="unbounded" minOccurs="0"/>
+ *       &lt;/sequence>
+ *       &lt;attribute name="copy-on-read" type="{http://www.w3.org/2001/XMLSchema}boolean" />
+ *       &lt;attribute name="is-server" type="{http://www.w3.org/2001/XMLSchema}boolean" />
+ *       &lt;attribute name="lock-timeout" type="{http://www.w3.org/2001/XMLSchema}string" />
+ *       &lt;attribute name="lock-lease" type="{http://www.w3.org/2001/XMLSchema}string" />
+ *       &lt;attribute name="message-sync-interval" type="{http://www.w3.org/2001/XMLSchema}string" />
+ *       &lt;attribute name="search-timeout" type="{http://www.w3.org/2001/XMLSchema}string" />
+ *       &lt;attribute name="version" use="required" type="{http://geode.apache.org/schema/cache}versionType" fixed="1.0" />
+ *     &lt;/restriction>
+ *   &lt;/complexContent>
+ * &lt;/complexType>
+ * </pre>
+ *
+ *
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "",
+    propOrder = {"cacheTransactionManager", "dynamicRegionFactory", "gatewayHub", "gatewaySender",
+        "gatewayReceiver", "gatewayConflictResolver", "asyncEventQueue", "cacheServer", "pool",
+        "diskStore", "pdx", "regionAttributes", "jndiBindings", "region", "functionService",
+        "resourceManager", "serializationRegistration", "backup", "initializer", "cacheElements"})
+@XmlRootElement(name = "cache", namespace = "http://geode.apache.org/schema/cache")
+public class CacheConfig {
```

Comment:
Can we call this `CacheConfigType` for consistency?

Reply:
Added a subtask under our make CC public story to move that entire package to a public package and consider renaming of its classes.